Import data cause BufferOverflowException

**Bug Description**
I am trying to import data to online db with hive table,  exception happens if there are some strings length bigger than 255: 

```log
Caused by: java.io.IOException: write row to openmldb failed on:  ... 
	at com._4paradigm.openmldb.spark.write.OpenmldbDataSingleWriter.write(OpenmldbDataSingleWriter.java:89)
	at com._4paradigm.openmldb.spark.write.OpenmldbDataSingleWriter.write(OpenmldbDataSingleWriter.java:39)
	at org.apache.spark.sql.execution.datasources.v2.DataWritingSparkTask$.$anonfun$run$1(WriteToDataSourceV2Exec.scala:419)
	at org.apache.spark.util.Utils$.tryWithSafeFinallyAndFailureCallbacks(Utils.scala:1496)
	at org.apache.spark.sql.execution.datasources.v2.DataWritingSparkTask$.run(WriteToDataSourceV2Exec.scala:457)
	at org.apache.spark.sql.execution.datasources.v2.V2TableWriteExec.$anonfun$writeWithV2$2(WriteToDataSourceV2Exec.scala:358)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:90)
	at org.apache.spark.scheduler.Task.run(Task.scala:131)
	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:506)
	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1462)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:509)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:750)
Caused by: java.nio.BufferOverflowException
	at java.nio.HeapByteBuffer.put(HeapByteBuffer.java:194)
	at java.nio.ByteBuffer.put(ByteBuffer.java:867)
	at com._4paradigm.openmldb.common.codec.FlexibleRowBuilder.build(FlexibleRowBuilder.java:385)
	at com._4paradigm.openmldb.sdk.impl.InsertPreparedStatementImpl.buildRow(InsertPreparedStatementImpl.java:302)
	at com._4paradigm.openmldb.sdk.impl.InsertPreparedStatementImpl.execute(InsertPreparedStatementImpl.java:317)
	at com._4paradigm.openmldb.spark.write.OpenmldbDataSingleWriter.write(OpenmldbDataSingleWriter.java:77)
	... 13 more
```

**Expected Behavior**
import data success

**Relation Case**
no


**Steps to Reproduce**

1. prepare some data that some all string value length bigger than 255 and some less
2. import these data to online db
3. exception with java.nio.BufferOverflowException

After digging into the code, 
```java 
    // FlexibleRowBuilder.java
    int totalSize = strFieldStartOffset + strAddrLen + strTotalLen;
    // check totalSize if bigger than UNIT8_MAX or UNIT16_MAX ...
    int curStrAddrSize = CodecUtil.getAddrLength(totalSize);
    if (curStrAddrSize > strAddrSize) {
        // strAddrBuf will be expanded if the totalSize bigger than UNIT8_MAX(255)
        strAddrBuf = expandStrLenBuf(curStrAddrSize, settedStrCnt);
        strAddrSize = curStrAddrSize;
        totalSize = strFieldStartOffset + strAddrLen + strTotalLen;
    }
```
private variable **strAddrBuf** will be expanded if totalSize bigger than UNIT8_MAX(255) and wil be used for the following records but never reduce the array size, that causes java.nio.BufferOverflowException.

currently i manually reduce the strAddrBuf size at the end of result allocate to solve the problem.

--- a/java/openmldb-common/src/main/java/com/_4paradigm/openmldb/common/codec/FlexibleRowBuilder.java
+++ b/java/openmldb-common/src/main/java/com/_4paradigm/openmldb/common/codec/FlexibleRowBuilder.java
@@ -392,6 +392,10 @@ public class FlexibleRowBuilder implements RowBuilder {
         result.put(baseFieldBuf.array());               // Base data type
         result.put(strAddrBuf.array());                 // String addr
         result.put(stringWriter.toByteArray());         // String value
+
+        // reset strAddrBuf
+        strAddrLen = metaData.getStrFieldCnt();
+        strAddrBuf = ByteBuffer.allocate(strAddrLen).order(ByteOrder.LITTLE_ENDIAN);
         return true;
     }
 

--- a/java/openmldb-common/src/main/java/com/_4paradigm/openmldb/common/codec/FlexibleRowBuilder.java
+++ b/java/openmldb-common/src/main/java/com/_4paradigm/openmldb/common/codec/FlexibleRowBuilder.java
@@ -390,12 +390,8 @@ public class FlexibleRowBuilder implements RowBuilder {
         result.putInt(totalSize);                       // Size
         result.put(nullBitmap.getBuffer());             // BitMap
         result.put(baseFieldBuf.array());               // Base data type
-        result.put(strAddrBuf.array());                 // String addr
+        result.put(strAddrBuf.array(), 0, strAddrLen);  // String addr
         result.put(stringWriter.toByteArray());         // String value
-
-        // reset strAddrBuf
-        strAddrLen = metaData.getStrFieldCnt();
-        strAddrBuf = ByteBuffer.allocate(strAddrLen).order(ByteOrder.LITTLE_ENDIAN);
         return true;
     }
 

--- a/java/openmldb-common/src/test/java/com/_4paradigm/openmldb/common/RowCodecTest.java
+++ b/java/openmldb-common/src/test/java/com/_4paradigm/openmldb/common/RowCodecTest.java
@@ -1018,5 +1018,41 @@ public class RowCodecTest {
             Assert.fail();
         }
     }
+
+    @Test
+    public void testLongString() {
+        try {
+            List<ColumnDesc> schema = new ArrayList<ColumnDesc>();
+            schema.add(ColumnDesc.newBuilder().setName("col1").setDataType(DataType.kInt).build());
+            schema.add(ColumnDesc.newBuilder().setName("col2").setDataType(DataType.kString).build());
+            schema.add(ColumnDesc.newBuilder().setName("col3").setDataType(DataType.kInt).build());
+            schema.add(ColumnDesc.newBuilder().setName("col4").setDataType(DataType.kString).build());
+            schema.add(ColumnDesc.newBuilder().setName("col5").setDataType(DataType.kString).build());
+            RowBuilder builder = new FlexibleRowBuilder(schema);
+            for (int i = 0; i < 20; i++) {
+                String str1 = genRandomString(100);
+                String str2 = i % 2 == 0 ? genRandomString(255) : genRandomString(20);
+                String str3 = i % 2 == 0 ? genRandomString(1000) : genRandomString(10);
+                Assert.assertTrue(builder.appendInt(1));
+                Assert.assertTrue(builder.appendString(str1));
+                Assert.assertTrue(builder.appendInt(10));
+                Assert.assertTrue(builder.appendString(str2));
+                Assert.assertTrue(builder.appendString(str3));
+                builder.build();
+
+                ByteBuffer buffer = builder.getValue();
+                RowView rowView = new RowView(schema, buffer, buffer.capacity());
+                Assert.assertEquals(rowView.getInt(0), new Integer(1));
+                Assert.assertEquals(rowView.getString(1), str1);
+                Assert.assertEquals(rowView.getString(3), str2);
+                Assert.assertEquals(rowView.getString(4), str3);
+
+                ((FlexibleRowBuilder)builder).clear();
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
+            Assert.assertTrue(false);
+        }
+    }
 }
 

